File: frontend/lib/models/my_ranking_info.dart
import 'package:client/models/ranking_info.dart';

class MyRankingInfo {
  final String distanceCategory;
  final int myRank;
  final int totalRankedCount;
  final double percentile;
  final RankingInfo myRecord; // 나의 상세 기록 정보를 포함

  MyRankingInfo({
    required this.distanceCategory,
    required this.myRank,
    required this.totalRankedCount,
    required this.percentile,
    required this.myRecord,
  });

  // JSON Map을 MyRankingInfo 객체로 변환하는 팩토리 생성자
  factory MyRankingInfo.fromJson(Map<String, dynamic> json) {
    return MyRankingInfo(
      distanceCategory: json['distanceCategory'],
      myRank: json['myRank'],
      totalRankedCount: json['totalRankedCount'],
      percentile: (json['percentile'] as num).toDouble(),
      myRecord: RankingInfo.fromJson(json['myRecord']),
    );
  }
}

File: backend/src/main/java/NPJ/Crewer/running/dto/response/MyRankingInfo.java
package NPJ.Crewer.running.dto.response;

import lombok.Getter;

@Getter
public class MyRankingInfo {
    private final String distanceCategory;
    private final int myRank;
    private final int totalRankedCount;
    private final double percentile;
    private final RankingInfo myRecord;

    public MyRankingInfo(String distanceCategory, int myRank, int totalRankedCount, RankingInfo myRecord) {
        this.distanceCategory = distanceCategory;
        this.myRank = myRank;
        this.totalRankedCount = totalRankedCount;
        this.percentile = (totalRankedCount > 0) ? ((double) myRank / totalRankedCount * 100) : 0;
        this.myRecord = myRecord;
    }
}
